pkg/image/apis/image/reference.go

```diff
@@ -0,0 +1,51 @@
+package image
+
+import (
+	"strings"
+
+	"github.com/openshift/origin/pkg/image/apis/image/internal/reference"
+)
+
+// NamedDockerImageReference points to a Docker image.
+type namedDockerImageReference struct {
+	Registry  string
+	Namespace string
+	Name      string
+	Tag       string
+	ID        string
+}
+
+// parseNamedDockerImageReference parses a Docker pull spec string into a
+// NamedDockerImageReference.
+func parseNamedDockerImageReference(spec string) (namedDockerImageReference, error) {
```

Comment:
@bparees the original in origin/pkg/image/reference is only used by the cyclical bit in source-to-image.  The original can be deleted after this merges and your bump happens.